bazel/defs.bzl

```diff
@@ -10,7 +10,63 @@
       (e.g. switch platforms)
     HOST_ARCHES is the set of support host architectures
       (e.g. for running against a simulator or testing)
+    STRATUM_DEFAULT_COPTS are the default flags passed to the compiler.
 """
-EMBEDDED_ARCHES = [ "x86" ]
-HOST_ARCHES = [ "x86" ]
-STRATUM_INTERNAL = [ "//stratum:__subpackages__" ]
\ No newline at end of file
+
+load(
+    "@com_google_absl//absl:copts/GENERATED_copts.bzl",
+    "ABSL_GCC_FLAGS",
+    "ABSL_GCC_TEST_FLAGS",
+    "ABSL_LLVM_FLAGS",
+    "ABSL_LLVM_TEST_FLAGS",
+)
+
+EMBEDDED_ARCHES = ["x86"]
+HOST_ARCHES = ["x86"]
+STRATUM_INTERNAL = ["//stratum:__subpackages__"]
+
+# Compiler warnings for different toolchains.
+STRATUM_DISABLED_COMPILER_WARNINGS_COMMON = [
+    "-Wno-sign-compare",
+    "-Wno-unused-function",
+    "-Wno-unused-parameter",
+    "-Wno-unused-variable",
+]
+STRATUM_DISABLED_COMPILER_WARNINGS_GCC = []
+STRATUM_DISABLED_COMPILER_WARNINGS_LLVM = [
+    "-Wno-gnu-zero-variadic-macro-arguments",
+    "-Wno-missing-variable-declarations",
+    "-Wno-suggest-destructor-override",
+    "-Wno-unused-lambda-capture",
+    "-Wno-unused-template",
+]
```

Comment:
I'm not super sold on the warnings, since these still apply to included external headers, and therefore produce a lot of text. But the output is much smaller for iterative cycles on the same files and might catch something there.

For egregious violations we still have the `-Werror` hammer.